testdata/quotes.zsh

```diff
@@ -0,0 +1,13 @@
+# Test quoting.
+
+ 'single'   'var: $var'   'subst: $(ls)'   'esc: \n \x01 \001'
+ "double"   "var: $var"   "subst: $(ls)"   "esc: \n \x01 \001" # TODO: these last 2 are not highlighted
+$'single'  $'var: $var'  $'subst: $(ls)'  $'esc: \n \x01 \001'
+ `batick`   `var: $var`   `subst: $(ls)`   `esc: \n \x01 \001`
+
+"'nest'" "`nest`"
+'"nest"' '`nest`'
+
+ "escape: \" \' \` \\ escape"
+ 'escape: \" \' \` \\ escape'  # TODO: maybe highlight \' as an error?
+$'escape: \" \' \` \\ escape'
```

Comment:
Careful here.  Since `\'` is perfectly valid syntax, this line has two single-quoted strings: one with contents `escape: \" \` and another with contents `  # TODO: maybe highlight \`.  There isn't any comment on this line at all, either.

Thus, if someone thinks that's a comment and adds the word "don't" to it, line 13 would no longer have a `$'…'` string in it.  That seems somewhat brittle.